_ml-commons-plugin/api/model-apis/register-model.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: default
+title: Register model
+parent: Model APIs
+grand_parent: ML Commons API
+nav_order: 10
+---
+
+# Register a model
+
+All versions of a particular model are held in a model group. You can either [register a model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-group-apis/register-model-group/) before registering a model to the group or register a first version of a model, thereby creating the group. Each model group name in the cluster must be globally unique. 
+
+If you are registering the first version of a model without first registering the model group, a new model group is created automatically with the following name and access level:
+
+- Name: The new model group will have the same name as the model. Because the model group name must be unique, ensure that your model name does not have the same name as any model groups in the cluster. 
+- Access level: The access level for the new model group is determined using the `access_mode`, `backend_roles`, and `add_all_backend_roles` parameters that you pass in the request. If you provide none of the three parameters, the new model group will be `private` if model access control is enabled on your cluster and `public` if model access control is disabled. The newly registered model is the first model version assigned to that model group. 
+
+Once a model group is created, provide its `model_group_id` to register a new model version to the model group. In this case, the model name does not need to be unique.
+
+If you're using [pretrained models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models#supported-pretrained-models) provided by OpenSearch, we recommend that you first register a model group with a unique name for these models. Then register the pretrained models as versions to that model group. This ensures that every model group has a globally unique model group name.
+{: .tip}
+
+For information about user access for this API, see [Model access control considerations]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/index/#model-access-control-considerations).
+
+If the model is more than 10 MB in size, ML Commons splits it into smaller chunks and saves those chunks in the model's index.
+
+## Path and HTTP methods
+
+```json
+POST /_plugins/_ml/models/_register
+```
+{% include copy-curl.html %}
+
+## Request fields
+
+All request fields are required. 
```

Comment:
This is not correct.
For different model, they need different fields. 

For pretrained model, check https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/, the mandatory fields are `name`, `version`, `model_format`

For uploading model via URL, refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/model_serving_framework/text_embedding_model_examples.md, mandatory fields:  

- `name`
- `version`
- `model_task_type`
- `model_format`
- `model_content_hash_value`
- `model_config`: {`model_type`, `embedding_dimension`, `framework_type`}, 
- `url`

Reply:
For remote model , refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/remote_inference.md